### PR TITLE
Move `CheckpointConnector.fault_tolerant_auto_save_path` out of `CheckpointConnector.hpc_resume_path`

### PR DESCRIPTION
**Pull request by jjenniferdai**
## What does this PR do?

Item 1. in https://github.com/PyTorchLightning/pytorch-lightning/issues/9407#issuecomment-991399437
Part of #9407 cleanup, move `CheckpointConnector.fault_tolerant_auto_save_path` out of `CheckpointConnector.hpc_resume_path`.

Note that unless we change line 165 here to `if self.hpc_resume_path or self.fault_tolerant_auto_save_path is not None: `, this means fault-tolerant checkpoints will no longer run `on_hpc_load` hook (deprecated but not yet removed).
I think this is the correct behavior, though lmk if you think line 165 should be edited to preserve the same behavior as before.
https://github.com/PyTorchLightning/pytorch-lightning/blob/3c4d06bd42ff3282a017cd9fa879bbd68e8568df/pytorch_lightning/trainer/connectors/checkpoint_connector.py#L163-L166

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/checkpoint_connector.py [hunk 1]
```diff
@@ -56,18 +56,22 @@ def hpc_resume_path(self) -> Optional[str]:
         max_version = self.max_ckpt_version_in_folder(dir_path_hpc, "hpc_ckpt_")
         if max_version is not None:
             return os.path.join(dir_path_hpc, f"hpc_ckpt_{max_version}.ckpt")
-        auto_save_checkpoint = os.path.join(dir_path_hpc, ".pl_auto_save.ckpt")
-        if os.path.exists(auto_save_checkpoint):
-            return auto_save_checkpoint
+
+    @property
+    def fault_tolerant_auto_save_path(self) -> Optional[str]:
```
Comment (carmocca) [inline pytorch_lightning/trainer/connectors/checkpoint_connector.py:61]: Make this protected?

**Review by carmocca — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/checkpoint_connector.py (hunk 1 above)
Comment (jjenniferdai) [reply pytorch_lightning/trainer/connectors/checkpoint_connector.py:61]: 👍 make `_hpc_resume_path` too?

Comment (awaelchli) [reply pytorch_lightning/trainer/connectors/checkpoint_connector.py]: hmm, but that was already public

Comment (jjenniferdai) [reply pytorch_lightning/trainer/connectors/checkpoint_connector.py]: hm yea, either protected or not I was thinking they should be consistent? (same auto-resume path concepts)

Comment (jjenniferdai) [reply pytorch_lightning/trainer/connectors/checkpoint_connector.py]: on a related note, rename `fault_tolerant_auto_save_path` --> `fault_tolerant_auto_resume_path` ?

Comment (carmocca) [reply pytorch_lightning/trainer/connectors/checkpoint_connector.py]: > hmm, but that was already public

@awaelchli would you say we should honor deprecation for public properties in connectors? (not mixins).
Wouldn't you consider them private as they are completely hidden from the user API?
One of their downsides is that they limit customization of the Trainer, which also means it's unlikely any users will have customized them.

If you agree with this and we don't deprecate, resolve this conversation so the PR gets merged :P

Comment (awaelchli) [reply pytorch_lightning/trainer/connectors/checkpoint_connector.py]: sorry for the delay, I didn't realize this depends on me.

**Pull request merged**